types/schemas/postSubmissionApplication/index.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  ApplicationType,
+  LandDrainageConsentApplicationType,
+  LDCApplicationType,
+  ListedApplicationType,
+  PAApplicationType,
+  PPApplicationType,
+  PrimaryApplicationType,
+} from '../prototypeApplication/enums/ApplicationType';
+import {PrototypeApplication} from '../prototypeApplication';
+import {PostSubmissionMetadata} from './Metadata';
+import {CaseOfficer} from './data/CaseOfficer';
+import {PublicComments, SpecialistComments} from './data/Comments';
+import {Submission} from './data/Submission';
+import {Validation} from './data/Validation';
+import {Consultation} from './data/Consultation';
+import {Assessment} from './data/Assessment';
+import {Appeal} from './data/Appeal';
+import {Application} from './data/Application';
+import {LocalPlanningAuthority} from './data/LocalPlanningAuthority';
+
+/**
+ * @internal
+ * The generic base type for all applications
+ * Takes a primary and granular application type which allows child properties to differ based on these inputs
+ * Deriving `TPrimary` from `TGranular` is possible in TS, but not in a way which is currently compatible with ts-json-schema-generator
+ *
+ * @todo create redacted form of PrototypeApplication
+ */
+interface ApplicationSpecification<
+  TPrimary extends PrimaryApplicationType,
+  TGranular extends ApplicationType,
+> {
+  applicationType: TGranular;
+  data: {
+    application: Application<TPrimary>;
+    localPlanningAuthority: LocalPlanningAuthority<TPrimary>;
+    submission: Submission<TPrimary>;
+    validation: Validation<TPrimary>;
+    consultation?: Consultation<TPrimary>;
+    assessment?: Assessment<TPrimary>;
+    appeal?: Appeal<TPrimary>;
+    caseOfficer: CaseOfficer<TPrimary>;
+  };
+  comments?: {
+    public?: PublicComments;
+    specialist?: SpecialistComments;
+  };
+  submission: PrototypeApplication;
+  metadata: PostSubmissionMetadata;
```

Comment:
I'm a little surprised to not see any mention of `files` at the root level (I see they can be associated with a comment). Are the original user-uploaded documents intentionally omitted/not necessary for a register?

(Perhaps aanother argument for different metadata again, as PlanX metadata send along required/recommended/optional files info!)

Reply:
Honestly I think I took them out and forgot to put them back in! 😂
We've got files all over the place now ones from submission, ones that are added to comments, some are added during appeal and theres also possibly versioning - so the nice neat files at the root level may not work so well anymore 🤔